src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -943,7 +943,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                 For Each tree As SyntaxTree In trees
                     If tree Is Nothing Then
-                        Throw New ArgumentNullException(String.Format(VBResources.Trees0, i))
+                        Throw New ArgumentNullException($"trees({i})")
```

Comment:
>$"trees({i})" [](http://example.com/codeflow?start=56&length=13)

Is this going to break localization?

Reply:
@AlekseyTs Yes. I think localization here does not make sense. We are passing `paramName` which shouldn't be translated. Usually in most cases, this is just `nameof(parameter)`. I'm not even sure if it's recommended to use `(<index>)` the way it's currently done. But at least it shouldn't be translated.

Reply:
> I think localization here does not make sense.

Well, the words in the resource string itself were translated. Therefore, I think a change like that doesn't simply fit under a "cleanup" theme. I suggest reverting changes like that.

Reply:
Extracted that part in https://github.com/dotnet/roslyn/pull/80116. Hopefully if https://github.com/dotnet/roslyn/pull/80116 can be merged soon, I'll just then merge main to this branch.

Reply:
@jcouv Yup. This PR should most likely be ready to merge too.